Editor/Scripts/BotManagement/RGBotSynchronizer.cs

```diff
@@ -31,24 +48,295 @@ private static void CreateNewBot()
+                        AssetDatabase.OpenAsset(newBot);
+                    }
+                }
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        private async Task CreateLocalBotRecordFromRG(RGBot newRemoteBot)
+        {
+            //Create directory
+            string botFolderPath = _this.CreateBotFolder(newRemoteBot.name, newRemoteBot.id);
+
+            string zipDownloadPath = ZipPathForBot(newRemoteBot.id);
+
+            if (File.Exists(zipDownloadPath))
+            {
+                File.Delete(zipDownloadPath);
+            }
+                    
+            //download and extract zip
+            await _rgServiceManager.DownloadBotCode(newRemoteBot.id,
+                zipDownloadPath,
+                () =>
+                {
+                    // unpack the zip to the bot folder - overwrite existing files
+                    ZipFile.ExtractToDirectory(zipDownloadPath, botFolderPath, true);
+                },
+                () => { }
+            );
             
-            // check for and create 'new' bots
+            //write asset record
+            _this.CreateBotAssetFile(botFolderPath, newRemoteBot.name, newRemoteBot.id);
+        }
+        
+        private async Task<List<RGBot>> GetBotsFromRG()
+        {
+            var remoteBots = new List<RGBot>();
+
+            // check for and create 'new' bots on regression games
+            await _rgServiceManager.GetBotsForCurrentUser(
+                (existingRGBots) =>
+                {
+                    remoteBots = existingRGBots
+                        .Where(v => v.IsUnityBot && v.IsLocal && v.codeSourceType == "ZIPFILE").ToList();
+                }, () => { }
+            );
+            return remoteBots;
+        }
+
+        private Dictionary<long, string> CreateLocalBotZipFiles(IEnumerable<RGBot> localBots)
+        {
+            var localMD5 = new Dictionary<long, string>();
+
+            // create zip files for all the bots I have locally
+            foreach (var rgBot in localBots)
+            {
+                var botId = rgBot.id;
+                var localBot = RGBotAssetsManager.GetInstance()?.GetBotAssetRecord(botId);
+                var md5 = ZipLocalBot(botId, localBot.Path);
+                localMD5[botId] = md5;
+            }
+
+            return localMD5;
+        }
+
+        private async Task<bool> UpdateBotZipOnRG(long botId, string md5)
+        {
+            var didUpdate = false;
+            // if zip checksums don't match.. push new zip files
+            // Get remote zip details, check md5sum, decide if need to push
+            var needsToUpload = false;
+
+            await _rgServiceManager.GetBotCodeDetails(
+                botId,
+                (botCodeDetails) =>
+                {
+                    if (md5 != botCodeDetails.md5)
```

Comment:
Forgot to put this in my other PR, do we need a migration for the existing zip bots? I don't really mind much considering no users... but curious about that.

Reply:
shouldn't need any migration.. this is computed on the fly in backend from reading the file stream itself, not saved in the DB